contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,333 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 #include "contrib/endpoints/src/api_manager/check_security_rules.h"
+#include <iostream>
+#include <sstream>
+#include "contrib/endpoints/src/api_manager/auth/lib/json_util.h"
+#include "contrib/endpoints/src/api_manager/proto/security_rules.pb.h"
+#include "contrib/endpoints/src/api_manager/utils/marshalling.h"
 
-#include <string>
-
-#include "contrib/endpoints/include/api_manager/api_manager.h"
-#include "contrib/endpoints/include/api_manager/request.h"
-
+using ::google::api_manager::auth::GetProperty;
+using ::google::api_manager::auth::GetStringValue;
 using ::google::api_manager::utils::Status;
+using ::google::protobuf::Map;
+using ::google::protobuf::util::error::Code;
 
 namespace google {
 namespace api_manager {
-
 namespace {
 
 const char kFirebaseServerStaging[] =
     "https://staging-firebaserules.sandbox.googleapis.com/";
 
+const char kFirebaseService[] =
+    "/google.firebase.rules.v1.FirebaseRulesService";
+
+const char kFailedFirebaseReleaseFetch[] = "Failed to fetch Firebase Release";
+const char kFailedFirebaseTest[] = "Failed to execute Firebase Test";
+const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
+const char kTestSuccess[] = "SUCCESS";
+
+void SetProtoValue(std::string key, ::google::protobuf::Value &value,
+                   ::google::protobuf::Value *head) {
+  ::google::protobuf::Struct *s = head->mutable_struct_value();
+  Map<std::string, google::protobuf::Value> *fields = s->mutable_fields();
+  (*fields)[key] = value;
+}
+
+const std::string GetReleaseName(
+    std::shared_ptr<context::RequestContext> request_context) {
```

Comment:
The CheckSecurityRules method is the entry point and it receives a share_ptr. Do you mean that I should extract the pointer from the shared pointer? I thought it is considered bad to extract a pointer out of a smart pointer. Lets discuss more.